src/System/Wlog/LoggerNameBox.hs

```diff
@@ -44,6 +45,15 @@ class HasLoggerName m where
     -- | Change logger name in context
     modifyLoggerName :: (LoggerName -> LoggerName) -> m a -> m a
 
+    default askLoggerName :: (MonadTrans t, t n ~ m, Monad n, HasLoggerName n) => m LoggerName
+    askLoggerName = lift askLoggerName
+
+    default modifyLoggerName :: (MFunctor t, t n ~ m, Monad n, HasLoggerName n)
+                             => (LoggerName -> LoggerName)
+                             -> m a
+                             -> m a
+    modifyLoggerName f = hoist (modifyLoggerName f)
+
 instance (Monad m, HasLoggerName m) => HasLoggerName (ReaderT a m) where
     askLoggerName = lift askLoggerName
```

Comment:
You should also remove all bodies of all boilerplate instances for `HasLoggerName` type class. The point of having default signatures is too make all instances shorter as possible and remove all boilerplate. In that way you can also verify that your default implementations are indeed correct. Maybe only for `ContT` monad your default signature won't work... But that's okay, usually nothing works for `ContT` and this monad should be avoided.

Reply:
Sorry, forgot about that.

Reply:
`ContT` monad is not working with default implementations, as you expected, so I left it as it is.